x/pairing/keeper/unresponsive_provider.go

```diff
@@ -86,141 +61,90 @@ func (k Keeper) getUnresponsiveProvidersToPunish(ctx sdk.Context, currentEpoch u
-	if !found {
-		// if providerCuCounterForUnreponsivenessEntry is not found, raise a flag to create new entry in providerCuCounterForUnreponsivenessMap
-		createNewproviderCuCounterForUnreponsivenessEntryFlag = true
+	// check which of the consts is larger
+	largerEpochsNumConst := epochsNumToCheckCUForComplainers
+	if epochsNumToCheckCUForUnresponsiveProvider > epochsNumToCheckCUForComplainers {
+		largerEpochsNumConst = epochsNumToCheckCUForUnresponsiveProvider
 	}
 
 	// count the CU serviced by the unersponsive provider and used CU of the complainers
-	for counter := uint64(0); counter < epochsNumToCheckCUForUnresponsiveProvider+epochsNumToCheckCUForComplainers; counter++ {
+	for counter := uint64(0); counter < largerEpochsNumConst; counter++ {
 		// get providerPaymentStorageKey for epochTemp (other traits from the stake entry)
 		providerPaymentStorageKey := k.GetProviderPaymentStorageKey(ctx, providerStakeEntry.GetChain(), epochTemp, sdkStakeEntryProviderAddress)
 
 		// try getting providerPaymentStorage using the providerPaymentStorageKey
 		providerPaymentStorage, found := k.GetProviderPaymentStorage(ctx, providerPaymentStorageKey)
-		if !found {
-			// Get previous epoch (from epochTemp)
-			previousEpoch, err := k.epochStorageKeeper.GetPreviousEpochStartForBlock(ctx, epochTemp)
-			if err != nil {
-				return utils.LavaError(ctx, k.Logger(ctx), "get_previous_epoch", map[string]string{"err": err.Error(), "epoch": fmt.Sprintf("%+v", epochTemp)}, "couldn't get previous epoch")
+		if found {
+			// counter is smaller than epochsNumToCheckCUForUnresponsiveProvider -> count CU serviced by the provider in the epoch
+			if counter < epochsNumToCheckCUForUnresponsiveProvider {
+				// count the CU by iterating through the uniquePaymentStorageClientProvider objects
+				for _, uniquePayment := range providerPaymentStorage.GetUniquePaymentStorageClientProvider() {
+					providerServicedCu += uniquePayment.GetUsedCU()
+				}
 			}
-			// update epochTemp
-			epochTemp = previousEpoch
 
-			// there is no providerPaymentStorage for this provider in this epoch -> continue searching
-			continue
-		}
-
-		// counter is smaller than epochsNumToCheckCUForUnresponsiveProvider -> only count CU serviced by the provider in the epoch
-		if counter < epochsNumToCheckCUForUnresponsiveProvider {
-			// count the CU by iterating through the uniquePaymentStorageClientProvider objects
-			cuCount := uint64(0)
-			for _, uniquePayment := range providerPaymentStorage.GetUniquePaymentStorageClientProvider() {
-				cuCount += uniquePayment.GetUsedCU()
-			}
+			// counter is smaller than epochsNumToCheckCUForComplainers -> count complainer CU
+			if counter < epochsNumToCheckCUForComplainers {
+				// Get servicersToPair param
+				servicersToPair, err := k.ServicersToPairCount(ctx, epoch)
```

Comment:
move it outside of the for loop and ask for the tempEpoch

Reply:
done